plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -626,29 +626,29 @@ public void onResponse(Map<Integer, MLOutput> multipleMLOutputs) {
                                             ignoreMissing,
                                             fullResponsePath
                                         );
-                                        Object modelOutputValuePerDoc;
-                                        if (modelOutputValue instanceof List
-                                            && ((List) modelOutputValue).size() == hitCountInPredictions.get(mappingIndex)) {
-                                            Object valuePerDoc = ((List) modelOutputValue)
-                                                .get(MapUtils.getCounter(writeOutputMapDocCounter, mappingIndex, modelOutputFieldName));
-                                            modelOutputValuePerDoc = valuePerDoc;
-                                        } else {
-                                            modelOutputValuePerDoc = modelOutputValue;
-                                        }
                                         // writing to search response extension
                                         if (newDocumentFieldName.startsWith(EXTENSION_PREFIX)) {
                                             Map<String, Object> params = ((MLInferenceSearchResponse) response).getParams();
                                             String paramsName = newDocumentFieldName.replaceFirst(EXTENSION_PREFIX + ".", "");
 
                                             if (params != null) {
-                                                params.put(paramsName, modelOutputValuePerDoc);
+                                                params.put(paramsName, modelOutputValue);
                                                 ((MLInferenceSearchResponse) response).setParams(params);
                                             } else {
                                                 Map<String, Object> newParams = new HashMap<>();
-                                                newParams.put(paramsName, modelOutputValuePerDoc);
+                                                newParams.put(paramsName, modelOutputValue);
                                                 ((MLInferenceSearchResponse) response).setParams(newParams);
                                             }
                                         } else {
+                                            Object modelOutputValuePerDoc;
+                                            if (modelOutputValue instanceof List
+                                                && ((List) modelOutputValue).size() == hitCountInPredictions.get(mappingIndex)) {
```

Comment:
If mappingIndex is not in hitCountInPredictions, you'll get a null → NullPointerException on size() == null.

Reply:
I will add this check